app/main.py

```diff
@@ -1,2 +1,82 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    """ each element in hash_table is considered as chain,
+        so we need check if there is node with key we get, using 'for' cycle
+        if it is - reassign/read/delete value/node
+        or if not - just append node to chain/raise KeyError"""
+    def __init__(self) -> None:
+        self._size = 0
+        self._capacity = 8
+        self._threshold = int(self._capacity * 2 / 3)
+        self._hash_table = [[] for _ in range(self._capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        hashed_key = hash(key)
+        index = hashed_key % self._capacity
+
+        if self._hash_table[index]:
+            for cell in self._hash_table[index]:
+                if cell[0] == key:
+                    cell[1] = value
+                    return
+
+        self._hash_table[index].append([key, value, hashed_key])
```

Comment:
Do you try to add a few elements in one cell in your dict? It's not how the dict works. Or did I not understand sth?

Reply:
![example](https://user-images.githubusercontent.com/71044367/196136191-48d96032-f258-4bc1-9e11-0d80ea449d35.PNG)
I've tried to implement kind of chain method for resolving collisions